packages/web/hooks/one-click-trading/use-one-click-trading-session.ts

```diff
@@ -22,42 +21,62 @@ export const useOneClickTradingSession = ({
   onExpire?: (params: { oneClickTradingInfo: OneClickTradingInfo }) => void;
 } = {}) => {
   const { accountStore, chainStore } = useStore();
-  const [isExpired, setIsExpired] = useState(false);
-  const { t } = useTranslation();
   const account = accountStore.getWallet(chainStore.osmosis.chainId);
   const featureFlags = useFeatureFlags();
 
-  const { value, loading } = useAsync(async () => {
-    const defaultReturn = {
-      info: undefined,
-      isEnabled: false,
-      isExpired: false,
-    };
-
-    if (!account?.address) {
-      return defaultReturn;
-    }
-
-    const info = await accountStore.getOneClickTradingInfo();
-    const isEnabled = await accountStore.isOneCLickTradingEnabled();
-    setIsExpired(await accountStore.isOneClickTradingExpired());
-
-    if (info?.userOsmoAddress !== account?.address) {
-      setIsExpired(false);
-      return defaultReturn;
-    }
-
-    return { info, isEnabled, isExpired };
-  }, [
-    accountStore,
-    isExpired,
-    accountStore.oneClickTradingInfo,
-    account?.address,
-  ]);
-
-  // Set a timeout to update the isExpired state
+  const [value, setValue] = useState<{
+    info: OneClickTradingInfo | undefined;
+    isEnabled: boolean;
+    isExpired: boolean;
+  }>({
+    info: undefined,
+    isEnabled: false,
+    isExpired: false,
+  });
+  const [loading, setLoading] = useState(true);
+
   useEffect(() => {
-    if (isExpired || !value?.info) return;
+    const disposer = reaction(
+      () => ({
+        address: account?.address,
+        oneClickTradingInfo: accountStore.oneClickTradingInfo,
+        useOneClickTrading: accountStore.useOneClickTrading,
+        isExpired: accountStore.hasOneClickTradingExpired,
+      }),
+      (data) => {
+        const defaultReturn = {
+          info: undefined,
+          isEnabled: false,
+          isExpired: false,
+        };
+
+        if (!data.address) {
+          setValue(defaultReturn);
+          setLoading(false);
+          return;
+        }
+
+        if (data.oneClickTradingInfo?.userOsmoAddress !== data.address) {
+          setValue(defaultReturn);
+        } else {
+          setValue({
+            info: data.oneClickTradingInfo,
+            isEnabled: data.useOneClickTrading,
+            isExpired: data.isExpired,
+          });
+        }
+        setLoading(false);
+      },
+      {
+        fireImmediately: true,
+      }
+    );
+
+    return () => disposer();
+  }, [account?.address, accountStore]);
```

Comment:
I prefer the previous solution because it doesn’t rely entirely on MobX. Since we plan to migrate the account store from MobX to React, retaining the previous syntax will be highly beneficial as it can help us avoid additional refactoring in the future. Do you think it's possible to keep that syntax or is using a `reaction` required?

Reply:
It's possible to keep this syntax and make the feature work see 70258a9060a0

I have left a note about how that flow is triggered in 2d4225083a5b 
and reverted the related changes (7f5e878bf785, 4fc31a0268d8) 

I wanted to create a robust fix because the wait loop felt hacky.
However, I also had concerns about whether that was the direction you wanted to take.
Thanks for clearing this up!